### PR TITLE
Add biking and driving transportation modes

### PR DESCRIPTION
**Pull request by andreeaceachir142**
# What 
- Add support for biking and driving transportation modes

# How 
- Adjust the `route-instructions-step` component to align with the design for driving and biking transportation modes
- Use the `mi-dropdown` component to contain the 3 supported transportation modes
- Add logic for changing the transportation mode and setting a new one whenever a value in the dropdown is selected 
- Add support for recalculating the route and updating the time and the distance based on the selected transportation mode 
- Adjust the dropdown component to align with the design 
- Create enum which holds the supported transportation modes

### CODE REVIEW

File: packages/components/src/components/dropdown/dropdown.scss [hunk 1]
```diff
@@ -93,10 +90,12 @@ mi-dropdown-item {
...
 
         &--from-inner-html {
-            display: flex;
-            flex-direction: row;
+            display: grid;
             align-items: center;
+            grid-auto-flow: column;
+            grid-template-columns: variables.$spacing-large 1fr;
             gap: variables.$spacing-x-small;
+            justify-items: center;
```
Comment (ammapspeople) [inline packages/components/src/components/dropdown/dropdown.scss:98]: Could we use `place-items`?

File: packages/components/src/components/dropdown/dropdown.scss [hunk 2]
```diff
@@ -213,7 +213,7 @@ mi-dropdown-item {
 }
 
 .empty-page {
-    > * {
+    >* {
```
Comment (ammapspeople) [inline packages/components/src/components/dropdown/dropdown.scss:216]:
```suggestion
    > * {
```

File: packages/components/src/components/dropdown/dropdown.scss [hunk 3]
```diff
@@ -247,10 +247,12 @@ mi-dropdown-item {
...
 
         &--from-inner-html {
-            display: flex;
-            flex-direction: row;
+            display: grid;
             align-items: center;
+            grid-auto-flow: column;
+            grid-template-columns: variables.$spacing-large 1fr;
             gap: variables.$spacing-x-small;
+            justify-items: center;
```
Comment (ammapspeople) [inline packages/components/src/components/dropdown/dropdown.scss:255]: `place-items`?

File: packages/map-template/src/components/Directions/Directions.jsx [hunk 4]
```diff
@@ -29,6 +33,8 @@ function Directions({ isOpen, onBack, directions }) {
     const [totalDistance, setTotalDistance] = useState();
     const [totalTime, setTotalTime] = useState();
 
+    const [travelMode, setTravelMode] = useState();
```
Comment (ammapspeople) [inline packages/map-template/src/components/Directions/Directions.jsx:36]: Is this necessary? Can't we use the provided prop as it is?

File: packages/map-template/src/components/Sidebar/Sidebar.jsx [hunk 5]
```diff
@@ -26,6 +27,9 @@ import LocationsList from '../LocationsList/LocationsList';
 function Sidebar({ currentLocation, setCurrentLocation, currentCategories, onLocationsFiltered, currentVenueName, directionsFromLocation, directionsToLocation, pushAppView, currentAppView, appViews, selectedMapType, filteredLocationsByExternalIDs, onLocationsFilteredByExternalIDs }) {
     const [directions, setDirections] = useState();
 
+    const [travelMode, setTravelMode] = useState(travelModes.WALKING);
+
```
Comment (ammapspeople) [inline packages/map-template/src/components/Sidebar/Sidebar.jsx:31]:
```suggestion
    const [travelMode, setTravelMode] = useState(travelModes.WALKING);
```

File: packages/map-template/src/components/Wayfinding/Wayfinding.jsx [hunk 6]
```diff
@@ -394,28 +416,49 @@ function Wayfinding({ onStartDirections, onBack, currentLocation, directionsToLo
...
+                        )}
+                        {hasGooglePlaces && <img className="wayfinding__google" alt="Powered by Google" src={GooglePlaces} />}
+                    </div>
+                </div>}
             {!searchTriggered && hasFoundRoute && !hasError && originLocation && destinationLocation && <div className={`wayfinding__details`} ref={detailsRef}>
-                <div className="wayfinding__accessibility">
-                    <input className="mi-toggle" type="checkbox" checked={accessibilityOn} onChange={e => setAccessibilityOn(e.target.checked)} />
-                    <div>Accessibility</div>
-                    <Tooltip text="Turn on Accessibility to get directions that avoids stairs and escalators."></Tooltip>
+                <div className="wayfinding__travel-details">
```
Comment (ammapspeople) [inline packages/map-template/src/components/Wayfinding/Wayfinding.jsx:433]:
```suggestion
                <div className="wayfinding__settings">
```
? 🤔

File: packages/map-template/src/components/WebComponentWrappers/Dropdown/Dropdown.jsx [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { useEffect, useRef } from 'react';
+
+/**
+ * React wrapper around the custom element <mi-dropdown>.
+ *
+ * @param {object} props
+ * @param {function} travelModeChanged - Function that is called when the travel mode is changed.
+ */
+function Dropdown({ travelModeChanged, children }) {
```
Comment (ammapspeople) [inline packages/map-template/src/components/WebComponentWrappers/Dropdown/Dropdown.jsx:9]: `travelModeChanged` seems too specific for a generic wrapper component.

File: packages/components/src/components/dropdown/dropdown.scss (hunk 3 above)
Comment (andreeaceachir142) [reply packages/components/src/components/dropdown/dropdown.scss]: Refactored the code to use `place-items`

File: packages/components/src/components/dropdown/dropdown.scss (hunk 1 above)
Comment (andreeaceachir142) [reply packages/components/src/components/dropdown/dropdown.scss]: Refactored the code to use `place-items`

File: packages/map-template/src/components/Wayfinding/Wayfinding.jsx (hunk 6 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/Wayfinding/Wayfinding.jsx]: Changed the className ✅

File: packages/map-template/src/components/WebComponentWrappers/Dropdown/Dropdown.jsx (hunk 7 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/WebComponentWrappers/Dropdown/Dropdown.jsx]: Renamed function inside dropdown wrapper component.

**Review by ammapspeople — APPROVED** [review]
✅

**Pull request merged**